TYPO3.Flow/Classes/TYPO3/Flow/Security/Cryptography/RsaWalletServicePhp.php

```diff
@@ -323,12 +342,25 @@ private function getModulus($keyResource)
      * @param string $cipher The ciphertext to decrypt
      * @param OpenSslRsaKey $privateKey The private key
      * @return string The decrypted plaintext
+     * @throws Exception
      */
     private function decryptWithPrivateKey($cipher, OpenSslRsaKey $privateKey)
     {
         $decrypted = '';
         $key = openssl_pkey_get_private($privateKey->getKeyString());
-        openssl_private_decrypt($cipher, $decrypted, $key);
+        if (openssl_private_decrypt($cipher, $decrypted, $key, $this->paddingAlgorithm) === false) {
+            // Fallback for data that was encrypted with old default OPENSSL_PKCS1_PADDING
+            if ($this->paddingAlgorithm !== OPENSSL_PKCS1_PADDING) {
```

Comment:
I don't understand that fallback mechanism. It won't be effective by default (i.e. paddingAlgorithm _is_ OPENSSL_PKCS1_PADDING)

Reply:
We used to encrypt using `OPENSSL_PKCS1_PADDING` - but change that default to `OPENSSL_PKCS1_OAEP_PADDING`. To be able to still decrypt existing data, we try with the old padding algorithm upon decryption failure, but only if the current padding is not the old default (because then the error cannot be caused by that change.)

Reply:
Ah right.. I tend to not read constant names, my bad